### ISSUE TITLE
Convert craftlib-chat to adventure

### ISSUE THREAD

**Issue by NickAcPT** [Zerite/CraftLib#19]
**Is your feature request related to a problem? Please describe.**
[Adventure](https://github.com/KyoriPowered/adventure) appears to be more established when it comes to dealing with chat messages. It has many features that this project can inherit from using it.

It abstracts Text components, boss-bars, sounds, and they even have an NBT API if you need (provided you also want to migrate from yours).

**Describe the solution you'd like**
Stop using `craftlib-chat` and move to `adventure`.

Although, I understand if you don't want to make the move into something made by another person/team.
Either way, thanks for the time and effort you put into this project!

**Comment by halotroop2288** [comment]
I'm going to make an attempt at implementing this in a PR. Wish me luck!

**Comment by sexnine** [comment]
Hey @Zerite / @KodingDev, could we please get your thoughts on this issue?

Are you guys open to switching to, or at least supporting [Adventure](https://github.com/KyoriPowered/adventure) for Chat?

If so, I can also have a go at implementing this and submitting a PR.
If this is something that you guys **would not** like, I won't bother.

**Comment by KodingDev** [comment]
This library was created mainly as a wholistic re-implementation of MC tech. From the beginning I didn't want to rely on any external libraries and rather create them all in-house. I stopped having time to maintain the library unfortunately, and - while I was going to focus on a rewrite in the years following - libraries such as adventure now replace a lot of the stuff I was planning to re-implement.

tldr; Don't worry about creating a PR to swap in adventure support, the lib has some core architectural flaws which need addressing regardless.

**Comment by sexnine** [comment]
> This library was created mainly as a wholistic re-implementation of MC tech. From the beginning I didn't want to rely on any external libraries and rather create them all in-house. I stopped having time to maintain the library unfortunately, and - while I was going to focus on a rewrite in the years following - libraries such as adventure now replace a lot of the stuff I was planning to re-implement.
> 
> tldr; Don't worry about creating a PR to swap in adventure support, the lib has some core architectural flaws which need addressing regardless.
@KodingDev 

That's no problem and I fully understand.  I've started using CraftLib in a project and it's been a really nice developer experience.

If you are interested, I'm happy to help rewrite whatever parts of the library need addressing and help maintain the library.
